parkbenchcommon/ramdisk.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2018 Joel Allen Luellwitz and Emily Frost
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+""" Provides a class for managing ramdisks."""
+
+__all__ = ['Ramdisk', 'RamdiskMountError']
+
+import logging
+import os
+import subprocess
+
+class RamdiskMountError(Exception):
+    """ Raised when a ramdisk mount operation fails."""
+
+class Ramdisk:
+    """ A class for managing ramdisks."""
+    def __init__(self, path, size, uid, gid, mode):
```

Comment:
Can these be named parameters? It seems like there is just too many of them.

Reply:
Actually, size, uid, gid, and mode should probably be moved to the mount method. I don't see a good reason for them to be set here.

Reply:
All arguments may be passed as named arguments. Arguments can be *required* to be named, if that's what you mean.

Reply:
I didn't realize that. That's fine.

Reply:
I ended up moving the arguments and simple integer validation to the size, uid, gid, and mode parameters.